resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -196,11 +196,17 @@ object YarnSparkHadoopUtil {
       .map(new Path(_).getFileSystem(hadoopConf))
       .toSet
 
+    // add the list of available namenodes for all namespaces in HDFS federation
+    val hadoopFilesystems = Option(hadoopConf.get("dfs.nameservices"))
+      .toSeq.flatMap(_.split(","))
+      .map(ns => hadoopConf.get(s"dfs.namenode.rpc-address.$ns"))
```

Comment:
if that namespace is listed in the `dfs.nameservices` config, this should exist, otherwise it is not a valid configuration. Shall we check for null in case we get an invalid config?

Reply:
I think it is good to check the nullable, at least it is no harm to the current code.